### PR TITLE
Understanding Cross-Site Scripting Attacks

### LINKED ISSUE #4686: Understanding Cross-Site Scripting Attacks
#### Proposed title of article
Understanding Cross-Site Scripting Attacks

#### Proposed article introduction
[Security]Cross-site scripting (XSS) is a web security flaw that allows an attacker to compromise user interactions with a susceptible application. It enables an attacker to get around the same-origin policy, which separates websites. Cross-site scripting flaws allow an attacker to impersonate a target user and do any actions or access any of the victim's data.

#### Key takeaways 
- Overview of Cross-Site Scripting Attacks
- Types of XSS 
- How to mitigate XSS attacks

#### Article quality
I'll be using a practical approach to explain.

#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by shosenwales**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure check

- [x]  Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4686 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your article for publishing.
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

@shosenwales

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers

is anyone available to jump on this PR ASAP.

Thank you

**Comment by john-amiscaray** [comment]
@hectorkambow I'm on it.

**Comment by john-amiscaray** [comment]
@shosenwales Good content. I thought it was pretty interesting. Let me know if there are any mistakes in my last commit or if there are any changes you are confused about. Be sure to go through my changes so you can understand some of the improvements/corrections I made so you may improve your future work!

**Comment by shosenwales** [comment]
Thanks @john-amiscaray . Will definitely go through

File: content/articles/understanding-cross-site-scripting-attacks/index.md
Comment (john-amiscaray) [inline content/articles/understanding-cross-site-scripting-attacks/index.md:16]: I forgot to comment on this part: please finish this sentence it seems you forgot something here

**Review by john-amiscaray — CHANGES_REQUESTED** [review]

File: content/articles/understanding-cross-site-scripting-attacks/index.md
Comment (shosenwales) [reply content/articles/understanding-cross-site-scripting-attacks/index.md:16]: hi @john-amiscaray 

i made a mistake of git pulling from master. I think i might have lost my work.
i cant even get into the branch.  Any idea what I can do

Comment (shosenwales) [reply content/articles/understanding-cross-site-scripting-attacks/index.md:16]: I have the article copied out tho.

Comment (shosenwales) [reply content/articles/understanding-cross-site-scripting-attacks/index.md]: Fixed. Kindly go through

**Comment by john-amiscaray** [comment]
@shosenwales Just went through the technical review and everything seemed fine. I also implemented some things I missed in the prelim review.

File: content/articles/understanding-cross-site-scripting-attacks/index.md
Comment (john-amiscaray) [inline content/articles/understanding-cross-site-scripting-attacks/index.md:65]: This sounds a little vague. I would add a further explanation to it.

File: content/articles/understanding-cross-site-scripting-attacks/index.md
Comment (john-amiscaray) [inline content/articles/understanding-cross-site-scripting-attacks/index.md:98]: Like what I said in the section about reflected XSS, this also sounds a little vague. I would try to give a more detailed explanation here if possible.

**Review by john-amiscaray — CHANGES_REQUESTED** [review]
I only have these two revisions for you to make. After that, we should be ready for a final review after the plagiarism check.

File: content/articles/understanding-cross-site-scripting-attacks/index.md
Comment (shosenwales) [reply content/articles/understanding-cross-site-scripting-attacks/index.md]: Fixed, kindly go through. Thanks

**Comment by john-amiscaray** [comment]
@shosenwales looking good. Just going to add the front matter and PRL now. From there we just need to wait for the plagiarism check and the final review.

**Comment by hectorkambow** [comment]
running a plagiarism check on this one now

**Comment by hectorkambow** [comment]
[Understanding Cross-Site Scripting Attacks #5170.pdf](https://github.com/section-engineering-education/engineering-education/files/7645930/Understanding.Cross-Site.Scripting.Attacks.5170.pdf)


👋 Good afternoon - and welcome to the EngEd program.
Upon running your article through our 3rd party plagiarism checker, it seemed to raise a few flags, and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page (within our new contributor folder in this Repo) to see more info on plagiarism and what is considered as such.

cc @john-amiscaray

**Comment by hectorkambow** [comment]
@shosenwales

**Comment by hectorkambow** [comment]
@shosenwales 
Lets take the time to make any needed changes - and review the content for any other issues that may need to be addressed as well. 

It seems this should be go to go after the plagiarism edits. 

Please be sure you are going over the entire article for any other mistakes or errors, as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility to properly review and proofread their work.

cc @john-amiscaray

**Comment by shosenwales** [comment]
@hectorkambow made some changes. Hopefully, it fixes it.

Kindly go through

**Comment by hectorkambow** [comment]
plagiarism check done
[Understanding Cross-Site Scripting Attacks #5170 second scan.pdf](https://github.com/section-engineering-education/engineering-education/files/7662799/Understanding.Cross-Site.Scripting.Attacks.5170.second.scan.pdf)
 
 
 @shosenwales  
 
 % was 14 - but I think it brought up sentences you added sources for. 
 
 Still would be wise to look through

**Comment by hectorkambow** [comment]
@john-amiscaray  

I believe you mentioned this PR is ready for a final?

**Comment by john-amiscaray** [comment]
> @john-amiscaray
> 
> I believe you mentioned this PR is ready for a final?

Yup, I was just waiting for the plagiarism check to be finished and for the proper edits to be made to remove any plagiarism.

**Comment by john-amiscaray** [comment]
@shosenwales Please change the introduction. The plagiarism check says the your intro is nearly identical to that of the source you used.

**Comment by john-amiscaray** [comment]
@hectorkambow I think we shouldn't add the **ready for final review** tag yet to let the author change the intro. Most of the intro was marked as identical to another source so I wouldn't say it's ready for a final review yet. I definitely think the intro should be revised.

**Comment by shosenwales** [comment]
Hi @john-amiscaray thanks for the review. Made some changes to the intro, kindly review

**Comment by john-amiscaray** [comment]
@shosenwales It seems there was a merge conflict we missed in this pull request. As a rule of thumb, run `git pull` in your local copy of the branch before you edit anything especially if you know that the reviewer has added a new commit. I fixed it however but keep this in mind for next time.

**Comment by john-amiscaray** [comment]
Should be ready for the final review now

**Comment by shosenwales** [comment]
Noted @john-amiscaray . Thanks

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**
